### PR TITLE
Change the default link and allow retry to handle 403 because of frequent visit

### PR DESCRIPTION
**Pull request by wznmickey**
1. Change from umjicanvas.com to jicanvas.com.
2. Allow retry in `clientGetJson`. Otherwise, there will be `403 Forbidden (Rate Limit Exceeded)` appearing.
`

### CODE REVIEW

**Comment by wznmickey** [comment]
自动回复：已收到，谢谢。

File: canvassyncer/__main__.py [hunk 1]
```diff
@@ -354,7 +361,7 @@ def initConfig():
     elif os.path.exists("./canvassyncer.json"):
         oldConfig = json.load(open("./canvassyncer.json"))
     print("Generating new config file...")
-    prevu = oldConfig.get("canvasURL", "") if oldConfig else "https://umjicanvas.com"
+    prevu = oldConfig.get("canvasURL", "") if oldConfig else "https://jicanvas.com"
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: should have another separated PR

File: canvassyncer/__main__.py [hunk 2]
```diff
@@ -84,7 +84,14 @@ async def close(self):
 
     async def clientGetJson(self, *args, **kwargs):
         async with self.sem:
-            resp = await self.client.get(*args, **kwargs)
+            check=0
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: need reformat

File: canvassyncer/__main__.py [hunk 3]
```diff
@@ -84,7 +84,14 @@ async def close(self):
...
+            while (check<=5):
+                resp = await self.client.get(*args, **kwargs)
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: retryable error may occur here

File: canvassyncer/__main__.py [hunk 4]
```diff
@@ -84,7 +84,14 @@ async def close(self):
...
+                try:
+                    _=resp.json()
+                    check = 10
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: just `break`

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: canvassyncer/__main__.py (hunk 1 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: Open in https://github.com/BoYanZh/Canvas-Syncer/pull/37

File: canvassyncer/__main__.py (hunk 4 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: directly return now

File: canvassyncer/__main__.py [hunk 5]
```diff
@@ -57,14 +57,20 @@ async def downloadMany(self, infos, totalSize=0):
...
+        retryTimes = 0
         checkError = bool(kwargs.pop("checkError", False))
-        async with self.sem:
-            resp = await self.client.get(*args, **kwargs)
-        res = resp.json()
-        if checkError and isinstance(res, dict) and res.get("errors"):
-            errMsg = res["errors"][0].get("message", "unknown error.")
-            print(f"\nError: {errMsg}")
-            exit(1)
+        while (retryTimes <= 5):
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: `()` can be removed.

File: canvassyncer/__main__.py [hunk 6]
```diff
@@ -57,14 +57,20 @@ async def downloadMany(self, infos, totalSize=0):
...
+            try:
+                async with self.sem:
+                    resp = await self.client.get(*args, **kwargs)
+                res = resp.json()
+                if checkError and isinstance(res, dict) and res.get("errors"):
+                    errMsg = res["errors"][0].get("message", "unknown error.")
+                    print(f"\nError: {errMsg}")
+                    exit(1)
+                return res
+            except Exception:
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: Better catch more detailed exception.

File: README.md [hunk 7]
```diff
@@ -39,7 +39,7 @@ Then input the information following the guide.
 
 *Note:*
 1. `courseCode` should be something like `VG100`, `ECE4530J`
-2. `courseID` should be an integer. Check the canvas link of the course. e.g. `courseID = 7` for <https://umjicanvas.com/courses/7>.
+2. `courseID` should be an integer. Check the canvas link of the course. e.g. `courseID = 7` for <https://jicanvas.com/courses/7>.
```
Comment (BoYanZh) [inline README.md]: Need to rebase on #36.

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: canvassyncer/__main__.py [hunk 8]
```diff
@@ -135,7 +144,7 @@ def prepareLocalFiles(self, courseID, folders):
     async def getCourseFoldersWithIDHelper(self, page, courseID):
         res = {}
         url = f"{self.baseUrl}/courses/{courseID}/folders?page={page}"
-        folders = await self.client.json(url)
+        folders = await self.client.json(url,debug=self.config["debug"])
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: need space after comma

**Review by BoYanZh — CHANGES_REQUESTED** [review]

**Review by BoYanZh — APPROVED** [review]

**Pull request merged**